src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerRemoveNote.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.hmcts.divorce.caseworker.event;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.sdk.api.CCDConfig;
+import uk.gov.hmcts.ccd.sdk.api.CaseDetails;
+import uk.gov.hmcts.ccd.sdk.api.ConfigBuilder;
+import uk.gov.hmcts.ccd.sdk.api.callback.AboutToStartOrSubmitResponse;
+import uk.gov.hmcts.divorce.common.ccd.PageBuilder;
+import uk.gov.hmcts.divorce.divorcecase.model.CaseData;
+import uk.gov.hmcts.divorce.divorcecase.model.State;
+import uk.gov.hmcts.divorce.divorcecase.model.UserRole;
+
+import java.util.Collections;
+
+import static uk.gov.hmcts.divorce.divorcecase.model.State.POST_SUBMISSION_STATES_WITH_WITHDRAWN_AND_REJECTED;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.CASE_WORKER;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.JUDGE;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.LEGAL_ADVISOR;
+import static uk.gov.hmcts.divorce.divorcecase.model.UserRole.SUPER_USER;
+import static uk.gov.hmcts.divorce.divorcecase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class CaseworkerRemoveNote implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String CASEWORKER_REMOVE_NOTE = "caseworker-remove-note";
+
+    public static final String ERROR_NOTE_ADDED = "Please use the add notes event if you would like to add new notes.";
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_REMOVE_NOTE)
+            .forStates(POST_SUBMISSION_STATES_WITH_WITHDRAWN_AND_REJECTED)
+            .name("Remove note")
+            .description("Remove case note")
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE_DELETE, SUPER_USER)
+            .grantHistoryOnly(CASE_WORKER, LEGAL_ADVISOR, JUDGE))
+            .page("removeNotes", this::midEvent)
+            .pageLabel("Remove case note")
+            .optional(CaseData::getNotes)
+            .done();
+    }
+
+    public AboutToStartOrSubmitResponse<CaseData, State> midEvent(final CaseDetails<CaseData, State> details,
+                                                                       final CaseDetails<CaseData, State> beforeDetails) {
+
+        final var beforeNotes = beforeDetails.getData().getNotes();
+        final var notes = details.getData().getNotes();
+        boolean hasAddedNotes = notes != null && (beforeNotes == null || beforeNotes.size() < notes.size());
```

Comment:
It is hard to protect against but users could add and remove a note within this event and this condition wouldn't catch that.

Reply:
Thanks, I didn't think it was worth complicating the validation to avoid that, as we are only making it available to superusers and would also have to block them from editing notes, which they could find useful to redact pieces of information without fully removing a note